Source/DataSources/KmlDataSource.js

```diff
@@ -1846,6 +2384,210 @@ define([
+                            JulianDate.addSeconds(now, minRefreshPeriod, date);
+                        }
+                        networkLink.time = date;
+                    } catch (e) {
+                        console.log('KML - NetworkLinkControl expires is not a valid date');
+                        remove = true;
+                    }
+                } else {
+                    console.log('KML - refreshMode of onExpire requires the NetworkLinkControl to have an expires element');
+                    remove = true;
+                }
+            }
+
+            var networkLinkEntity = networkLink.entity;
+            var entityCollection = dataSource._entityCollection;
+            var newEntities = newEntityCollection.values;
+
+            function removeChildren(entity) {
+                entityCollection.remove(entity);
+                var children = entity._children;
+                var count = children.length;
+                for(var i=0;i<count;++i) {
+                    removeChildren(children[i]);
+                }
+            }
+
+            // Remove old entities
+            var entitiesCopy = entityCollection.values.slice();
+            for (var i=0;i<entitiesCopy.length;++i) {
+                if (entitiesCopy[i].parent === networkLinkEntity) {
+                    entitiesCopy[i].parent = undefined;
+                    removeChildren(entitiesCopy[i]);
+                }
+            }
+
+            // Add new entities
+            entityCollection.suspendEvents();
+            for (i = 0; i < newEntities.length; i++) {
+                if (!defined(newEntities[i].parent)) {
+                    newEntities[i].parent = networkLinkEntity;
+                }
+                entityCollection.add(newEntities[i]);
+            }
+            entityCollection.resumeEvents();
+
+            // No refresh information remove it, otherwise update lastUpdate time
+            if (remove) {
+                networkLinks.splice(networkLinks.indexOf(networkLink), 1);
+            } else {
+                networkLink.lastUpdated = now;
+            }
+
+            var availability = entityCollection.computeAvailability();
+
+            var start = availability.start;
+            var stop = availability.stop;
+            var isMinStart = JulianDate.equals(start, Iso8601.MINIMUM_VALUE);
+            var isMaxStop = JulianDate.equals(stop, Iso8601.MAXIMUM_VALUE);
+            if (!isMinStart || !isMaxStop) {
+                var clock = dataSource._clock;
+
+                if (clock.startTime !== start ||  clock.stopTime !== stop) {
+                    clock.startTime = start;
+                    clock.stopTime = stop;
+                    dataSource._changed.raiseEvent(dataSource);
+                }
+            }
+
+            networkLink.updating = false;
+            networkLink.needsUpdate = false;
+            dataSource._refresh.raiseEvent(dataSource, networkLink.href);
```

Comment:
Should we raise this here, or wait until we are done processing all links?  Do you know what other implementation do?